nominal/core/dataset.py

```diff
@@ -209,6 +210,49 @@ def add_to_dataset_from_io(
         )
         self._clients.ingest.trigger_file_ingest(self._clients.auth_header, request)
 
+    def add_mcap_to_dataset(
+        self,
+        path: Path | str,
+        include_topics: Iterable[str] | None = None,
+        exclude_topics: Iterable[str] | None = None,
+    ) -> None:
+        """Add an MCAP file to an existing dataset."""
+        self.poll_until_ingestion_completed()
+        mcap_path = Path(path)
+        s3_path = upload_multipart_file(
+            self._clients.auth_header,
+            mcap_path,
+            self._clients.upload,
+            file_type=FileTypes.MCAP,
+        )
+        source = ingest_api.IngestSource(s3=ingest_api.S3IngestSource(path=s3_path))
+
+        channels = ingest_api.McapChannels(all=api.Empty())
+        if include_topics is not None and exclude_topics is not None:
+            include_topics = [t for t in include_topics if t not in exclude_topics]
+        if include_topics is not None:
+            channels = ingest_api.McapChannels(
+                include=[api.McapChannelLocator(topic=topic) for topic in include_topics]
+            )
+        elif exclude_topics is not None:
+            channels = ingest_api.McapChannels(
+                exclude=[api.McapChannelLocator(topic=topic) for topic in exclude_topics]
+            )
+
+        request = ingest_api.IngestRequest(
```

Comment:
might also be able to lump this into a shared function to create the request, parameterized by the target?